### PR TITLE
Add a prebuilt ICU Analyzer

### PR DESCRIPTION
**Pull request by romseygeek**
The ICU plugin provides the building blocks of an analysis chain, but doesn't actually have a prebuilt analyzer.  It would be a better for users if there was a simple analyzer that they could use out of the box, and also something we can point to from the CJK Analyzer docs as a superior alternative.

Relates to #34285

### CODE REVIEW

File: docs/reference/analysis/analyzers/lang-analyzer.asciidoc [hunk 1]
```diff
@@ -421,6 +421,9 @@ PUT /catalan_example
 [[cjk-analyzer]]
 ===== `cjk` analyzer
 
+NOTE: You may find that `icu_analyzer` in the ICU analysis plugin works better
+for CJK text than the `cjk` analyzer.
```
Comment (nik9000) [inline docs/reference/analysis/analyzers/lang-analyzer.asciidoc]: Maybe add something like "experiment with your text and queries" or something like that.

File: plugins/analysis-icu/src/main/java/org/elasticsearch/index/analysis/IcuAnalyzerProvider.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+public class IcuAnalyzerProvider extends AbstractIndexAnalyzerProvider<Analyzer> {
+
+    private final Normalizer2 normalizer;
+
+    public IcuAnalyzerProvider(IndexSettings indexSettings, Environment environment, String name, Settings settings) {
+        super(indexSettings, name, settings);
+        String method = settings.get("method", "nfkc_cf");
+        String mode = settings.get("mode");
+        if (!"compose".equals(mode) && !"decompose".equals(mode)) {
+            mode = "compose";
```
Comment (nik9000) [inline plugins/analysis-icu/src/main/java/org/elasticsearch/index/analysis/IcuAnalyzerProvider.java]: I'd prefer to throw an exception if they pass a string that isn't one of the options.

Comment (nik9000) [reply plugins/analysis-icu/src/main/java/org/elasticsearch/index/analysis/IcuAnalyzerProvider.java]: And just default to `compose` if they don't pass anything.

**Review by nik9000 — CHANGES_REQUESTED** [review]
I left a small thing about the default, otherwise I like the idea a lot!

**Review by nik9000 — APPROVED** [review]
LGTM

**Comment by romseygeek** [comment]
@elasticmachine run docs test please

**Comment by romseygeek** [comment]
@elasticmachine test this please

**Pull request merged**